### PR TITLE
feat(vcs): bubble up SSH key conversion error for better debugging experience

### LINKED ISSUE #914: Import git ssh error silenced
The SSH key conversion does not surface the error: https://github.com/grafana/alloy/blob/main/internal/vcs/auth.go#L29

When this happens, the given ssh key will be ignored and the go-git will try to use the ssh-agent instead: https://github.com/go-git/go-git/blob/master/plumbing/transport/ssh/common.go#L39

This is not good:

- if the key that I pass is wrong but the ssh-agent is correctly setup, then it will work but I won't know that I passed wrong information. It's also not using the auth method that I wanted.
- if the key is wrong and the ssh-agent is not set, then I will get an error related to the ssh-agent. I won't know what's wrong my ssh key which makes it super hard to debug

### PR DESCRIPTION
**Pull request by hainenber**
#### Which issue(s) this PR fixes

Closes #914 

#### PR Checklist

- [x] CHANGELOG.md updated

### CODE REVIEW

File: internal/vcs/auth.go [hunk 1]
```diff
@@ -16,20 +16,23 @@ type GitAuthConfig struct {
...
 
 	if h.BasicAuth != nil {
-		return h.BasicAuth.Convert()
+		return h.BasicAuth.Convert(), nil
 	}
 
 	if h.SSHKey != nil {
-		key, _ := h.SSHKey.Convert()
-		return key
+		key, err := h.SSHKey.Convert()
```
Comment (wildum) [inline internal/vcs/auth.go:29]: can be simplified by having directly `return h.SSHKey.Convert()` here

File: CHANGELOG.md [hunk 2]
```diff
@@ -91,6 +91,8 @@ Main (unreleased)
 
 - Updated Prometheus dependency to [v2.51.2](https://github.com/prometheus/prometheus/releases/tag/v2.51.2) (@thampiotr)
 
+- Bubble up SSH key conversion error to facilitate failed `import.git`. (@hainenber)
+
```
Comment (wildum) [inline CHANGELOG.md:95]: I would actually put it in bug fixes

**Review by wildum — COMMENTED** [review]
Thanks for taking care of this issue!

File: internal/vcs/auth.go (hunk 1 above)
Comment (hainenber) [reply internal/vcs/auth.go]: Done in HEAD :D

**Pull request merged**